### PR TITLE
Implementation of `MMIO` update

### PR DESCRIPTION
**Pull request by OlivierBBB**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -82,7 +82,10 @@ LOG_INFO := loginfo
 
 MMU := mmu
 
-MMIO := mmio/columns.lisp #TODO enable the MMIO constraint and lookup
+MMIO := mmio/columns.lisp
+#TODO enable the MMIO constraint and lookup
```
Comment (letypequividelespoubelles) [inline Makefile:86]: why did you change the std way ?

**Review by letypequividelespoubelles — COMMENTED** [review]
impossible to see what you changed, too many formatting change

File: mmio/instructions.lisp [hunk 2]
```diff
@@ -74,7 +74,17 @@
          (eq! INDEX_A SLO)
          (eq! VAL_A_NEW VAL_A)
          (eq! INDEX_X TLO)
-         (one-to-one-padded LIMB BYTE_A [ACC 1] [POW_256 1] SBO SIZE [BIT 1] [BIT 2] [BIT 3] CT)))
+         (one-to-one-padded    LIMB
+                               BYTE_A
+                               [ACC 1]
+                               [POW_256 1]
+                               SBO
+                               TBO
```
Comment (OlivierBBB) [inline mmio/instructions.lisp]: Addition of `TBO`

File: mmio/patterns.lisp [hunk 3]
```diff
@@ -24,15 +24,25 @@
...
-         (power pow bit3 counter)
-         (if-eq counter LLARGEMO
-                (eq! target (* accumulator pow)))))
+;; [1 => 1 Padded]
+(defpurefun    (one-to-one-padded    target
+                                     source_byte
+                                     accumulator
+                                     pow
+                                     source_marker
+                                     target_marker
```
Comment (OlivierBBB) [inline mmio/patterns.lisp:33]: Addition of `target_marker`

File: mmio/patterns.lisp [hunk 4]
```diff
@@ -24,15 +24,25 @@
...
+                                     size
+                                     bit1
+                                     bit2
+                                     bit3
+                                     counter)
+               (begin (plateau bit1 source_marker counter)
+                      (plateau bit2 (+ source_marker size) counter)
+                      (plateau bit3 (+ target_marker size) counter)
```
Comment (OlivierBBB) [inline mmio/patterns.lisp:41]: usage here

File: mmio/patterns.lisp [hunk 5]
```diff
@@ -45,33 +45,34 @@
...
-                               bit4
+(defpurefun    (two-to-one-padded    target
+                                     source1_byte
+                                     source2_byte
+                                     accumulator1
+                                     accumulator2
+                                     pow1
+                                     pow2
+                                     source1_marker
+                                     target_marker
```
Comment (OlivierBBB) [inline mmio/patterns.lisp:56]: Addition of `target_marker`

File: mmio/instructions.lisp [hunk 6]
```diff
@@ -95,7 +95,21 @@
...
-         (two-to-one-padded LIMB BYTE_A BYTE_B [ACC 1] [ACC 2] [POW_256 1] [POW_256 2] SBO SIZE [BIT 1] [BIT 2] [BIT 3] [BIT 4] CT)))
+         (two-to-one-padded LIMB
+                            BYTE_A
+                            BYTE_B
+                            [ACC 1]
+                            [ACC 2]
+                            [POW_256 1]
+                            [POW_256 2]
+                            SBO
+                            TBO
```
Comment (OlivierBBB) [inline mmio/instructions.lisp]: `TBO`

File: mmio/patterns.lisp [hunk 7]
```diff
@@ -65,14 +65,14 @@
...
                       (plateau bit3 (- (+ target_marker LLARGE) source1_marker)) counter)
-                      (plateau bit4 size counter)
-                      (isolate-suffix accumulator1 source1_byte bit1 counter)
-                      (isolate-prefix accumulator2 source2_byte bit2 counter)
-                      (power pow1 bit3 counter)
-                      (power pow2 bit4 counter)
-                      (if-eq counter LLARGEMO
-                             (eq! target
-                                  (+ (* accumulator1 pow1) (* accumulator2 pow2))))))
+               (plateau bit4 (+ target_marker size) counter)
```
Comment (OlivierBBB) [inline mmio/patterns.lisp:72]: `target_marker`

File: Makefile (hunk 1 above)
Comment (OlivierBBB) [reply Makefile:86]: You mean why did I move the comment ? I don't know
